qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,680 @@
+
+        Raises:
+            QiskitError:
+                - When __x_key__ is not defined in the circuit metadata.
+        """
+        if self.__series__:
+            series = self.__series__
+        else:
+            series = [
+                SeriesDef(
+                    name="fit-curve-0",
+                    param_names=self.__param_names__,
+                    fit_func_index=0,
+                    filter_kwargs=None,
+                )
+            ]
+
+        def _is_target_series(datum, **filters):
+            try:
+                return all(datum["metadata"][key] == val for key, val in filters.items())
+            except KeyError:
+                return False
+
+        curve_data = list()
+        for curve_properties in series:
+            if curve_properties.filter_kwargs:
+                # filter data
+                series_data = [
+                    datum
+                    for datum in experiment_data.data
+                    if _is_target_series(datum, **curve_properties.filter_kwargs)
+                ]
+            else:
+                # use data as-is
+                series_data = experiment_data.data
+
+            # Format x, y, yerr data
+            try:
+                xvals = np.asarray(
+                    [datum["metadata"][self.__x_key__] for datum in series_data], dtype=float
+                )
+            except KeyError as ex:
+                raise QiskitError(
+                    f"X value key {self.__x_key__} is not defined in circuit metadata."
+                ) from ex
+            yvals, sigmas = self._data_processing(series_data)
+
+            # Get common metadata fields except for xval and filter args.
+            # These properties are obvious.
+            common_keys = list(
+                functools.reduce(
+                    lambda k1, k2: k1 & k2,
+                    map(lambda d: d.keys(), [datum["metadata"] for datum in series_data]),
+                )
+            )
+            common_keys.remove(self.__x_key__)
+            if curve_properties.filter_kwargs:
+                for key in curve_properties.filter_kwargs:
+                    common_keys.remove(key)
+
+            # Extract common metadata for the curve
+            curve_metadata = defaultdict(set)
+            for datum in series_data:
+                for key in common_keys:
+                    curve_metadata[key].add(datum["metadata"][key])
+
+            curve_data.append(
+                CurveEntry(
+                    curve_name=curve_properties.name,
+                    x_values=xvals,
+                    y_values=yvals,
+                    y_sigmas=sigmas,
+                    metadata=dict(curve_metadata),
+                )
+            )
+
+        return curve_data
+
+    def _series_curve_fit(
+        self,
+        curve_data: List[CurveEntry],
+        p0: Optional[np.ndarray] = None,
```

Comment:
I don't think passing p0 as a number list is good enough. In interleaved RB, I compute the p0 from the curve data (i.e. the xdata and ydata with the additional parameter of the number of qubits);  I think we need to allow p0 here to be a function which gets the formatted data and an optional parameter set for it.

Reply:
Thanks @gadial . This is good point. I prepared `_setup_fitting` function that allows you to take input data with some user input **options to create a list of `FitOptions` dictionary. a3dd099